party_time/party_time.lua

```diff
@@ -0,0 +1,404 @@
+    options.default = option
+    settings.save(options)
+end
+
+pt:register('defualt', default_handler, '<option:lookup_option>')
+
+-- Packet Event Handlers
+-- Recieve Invites & Recieve Requests
+local default_handlers = {
+    invite = {
+        ask = function(name)
+            invite_dialog = {
+                state = invite_dialog_state,
+                add_to_whitelist = false,
+                name = name,
+            }
+            unhandled_invite = true
+        end,
+        whitelist = function()
+            coroutine.schedule(function()
+                local clock = os.clock()
+                repeat
+                    if (os.clock() - clock) > 90 then
+                        return
+                    end
+                    coroutine.sleep_frame()
+                until(#treasure == 0)
+                command.input('/join')
+            end)
+        end,
+        blacklist = function()
+            command.input('/decline')
+        end
+    },
+    request = { 
+        ask = function(name)
+            unhandled_requests[name] = {
+                state = {
+                    title = 'Party Request',
+                    style = 'normal',
+                    x = options.ui.x,
+                    y = options.ui.y,
+                    width = 179,
+                    height = 96,
+                    resizable = false,
+                    moveable = true,
+                    closable = true,
+                },
+                add_to_whitelist = false,
+            }
+        end,
+        whitelist = function(name)
+            command.input('/pcmd add '.. name)
+        end,
+        blacklist = function()
+            return
+        end,
+    }
+}
+
+packets.incoming[0x0DC]:register(function(p)
+    if options.auto.accept and options.whitelist:contains(p.player_name) then
+        coroutine.schedule(function()
+            local clock = os.clock()
+            repeat
+                if (os.clock() - clock) > 90 then
+                    return
+                end
+                coroutine.sleep_frame()
+            until(#treasure == 0)
+            command.input('/join')
+        end)
+    elseif options.auto.decline and options.blacklist:contains(p.player_name) then
+        command.input('/decline')
+    else
+        default.invite[options.default](p.player_name)
+    end
+end)
+
+packets.incoming[0x11D]:register(function(p)
+    if options.auto.accept and options.whitelist:contains(p.player_name) then
+        command.input('/pcmd add '..p.player_name)
+    elseif options.auto.decline and options.blacklist:contains(p.player_name) then
+        return --ignore request by providing a dialog to user
+    else
+        default.request[options.default](p.player_name)
+    end
+end)
+
+packets.outgoing[0x074]:register(function(p)
+    unhandled_invite = false
+end)
+
+-- User Interface for Accepting|Declining
+-- Pop-Up menus Invites & Requests
+ui.display(function()
+    if options.ui.enabled then
```

Comment:
I'm a bit of a fan of reduced nesting by "failing early", or "returning early" in this case. Since this `if` block spans the entire function, I prefer to invert it:
```lua
ui.display(function()
    if not options.ui.enabled then
        return
    end

    --[[ rest here, saves us 4 spaces x n lines! ]]
end)
```

Not that it really matters, nor is this a request, just a comment.